android/anc/src/main/java/org/smartregister/fhircore/anc/data/AncPaginatedDataSource.kt

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2021 Ona Systems, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.smartregister.fhircore.anc.data
+
+import com.google.android.fhir.FhirEngine
+import com.google.android.fhir.search.Search
+import com.google.android.fhir.search.StringFilterModifier
+import org.hl7.fhir.r4.model.Patient
+import org.smartregister.fhircore.anc.data.model.AncItem
+import org.smartregister.fhircore.engine.data.domain.util.DomainMapper
+import org.smartregister.fhircore.engine.data.domain.util.PaginatedDataSource
+
+class AncPaginatedDataSource(
```

Comment:
This can be renamed to `AncPatientPaginatedDataSource` as there is a possibility to have multiple paginated data sources for ANC application.